### PR TITLE
No Results Found

### LINKED ISSUE #147: [Access] [FE] Indicate no results found on email search
## Context

Background information relevant to the ticket

## Requirements

TBD - something like this? https://ant.design/components/empty

Fix #148 

#### PM

- [ ] Product-specific requirements

#### EM

- [ ] Engineering-specific requirements

#### Designer

- [ ] Design-specific requirements

### LINKED ISSUE #148: [BUG] Text/Visual Display for No Results on Email Search
## Description

#214 

When you search for an email on the access management page and there are no results, the page is just blank. There should be a place holder making it more obvious that there are no results

### LINKED ISSUE #214: [BUG]  Edge Case: no search results
## Description

We need to have edge cases for what the result div will display when there is no search results for:
1. Volunteer Search
2. Access Management
3. Feed

### PR DESCRIPTION
**Pull request by uma-anand**
## No Results Found

Issue Number(s): #214 #147 #148 

What does this PR change and why?

### Checklist

- [ ] Requirements have been implemented
- [ ] Acceptance criteria are met
- [ ] Database schema docs have been updated or are not necessary
- [ ] Code follows design and style guidelines
- [ ] Commits follow guidelines (concise, squashed, etc)
- [x] Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### Critical Changes

- Database change/migration to run
- Environment config change
- Breaking API change

### Related PRs

- #number_of_pr

### Testing

Enumerate steps to test the functionality of your ticket. This should include edge cases and testing of error handling, if applicable.

### CODE REVIEW

File: web/components/Feed/Feed.tsx [hunk 1]
```diff
@@ -607,20 +607,26 @@ function Feed(props: {
...
-                      onPostViewOpen();
-                    }}
-                    _hover={{ cursor: "pointer" }}
-                    key={p._id.toString()}
-                  >
-                    <FeedPostCard post={p} />
-                  </Box>
-                );
-              })}
+              {debouncedFeedPosts && debouncedFeedPosts.length != 0 ? (
```
Comment (katsukixyz) [inline web/components/Feed/Feed.tsx]: Use optional chaining, `debouncedFeedPosts?.length`. Use `>` instead of `!=`

File: web/pages/access.tsx [hunk 2]
```diff
@@ -57,6 +57,11 @@ function Access() {
           search={search}
           setSearch={setSearch}
         />
+        {!accounts.isLoading &&
+          !isUpdating &&
+          (!accounts.data || accounts.data.length == 0) && (
+            <Text> No results found.</Text>
+          )}
```
Comment (katsukixyz) [inline web/pages/access.tsx]: Move this into `AccountTable`

File: web/components/Users/Results.tsx [hunk 3]
```diff
@@ -115,6 +115,9 @@ export default function Results({ filters, setSearched }: ResultsProps) {
         )}
         <Divider />
       </Stack>
+      {(!users || !users.data || users.data.length == 0) && (
+        <Text align={"center"}>No results found.</Text>
+      )}
```
Comment (katsukixyz) [inline web/components/Users/Results.tsx]: This ternary should be modified so that we render the below `Flex` block only when there is data

**Review by katsukixyz — CHANGES_REQUESTED** [review]

File: web/components/AccessManagement/AccountTable.tsx [hunk 4]
```diff
@@ -36,36 +36,40 @@ function AccountTable(props: PropertyType) {
         updateItemsToDelete={updateItemsToDelete}
       ></TableHeader>
       {!isLoading ? (
```
Comment (katsukixyz) [inline web/components/AccessManagement/AccountTable.tsx:38]: `isLoading: true` holistically determines the spinner here, so define it before everything else with just `isLoading && <Spinner />`. Then you only have to check for `accountList?.length` so you won't need the nested ternary

File: web/components/Feed/Feed.tsx [hunk 5]
```diff
@@ -607,20 +607,26 @@ function Feed(props: {
...
-                      onPostViewOpen();
-                    }}
-                    _hover={{ cursor: "pointer" }}
-                    key={p._id.toString()}
-                  >
-                    <FeedPostCard post={p} />
-                  </Box>
-                );
-              })}
+              {(debouncedFeedPosts?.length ?? 0) > 0 ? (
```
Comment (katsukixyz) [inline web/components/Feed/Feed.tsx]: Sorry, I think I misspoke earlier - keep the `debouncedFeedPosts && debouncedFeedPosts > 0` condition from earlier. I'm so sorry about that 😭

**Review by katsukixyz — CHANGES_REQUESTED** [review]

File: web/components/Users/Results.tsx [hunk 6]
```diff
@@ -115,56 +115,65 @@ export default function Results({ filters, setSearched }: ResultsProps) {
...
-              padding={2}
-              h="fit-content"
-            >
-              <Grid
-                templateColumns={{
-                  base: "0 max-content",
-                  sm: "1fr max-content",
-                }}
-                gap={2}
+      {users?.data?.length ?? 0 > 0 ? (
```
Comment (katsukixyz) [inline web/components/Users/Results.tsx]: `users?.data && users.data.length > 0` here, sorry!

File: web/components/AccessManagement/AccountTable.tsx (hunk 4 above)
Comment (uma-anand) [reply web/components/AccessManagement/AccountTable.tsx:38]: So I tried that but sometimes while the spinner is still loading (especially close to when the spinner stops spinning), it displays the account list or the no results found along with the spinner- and I wanted to avoid that.

Comment (uma-anand) [reply web/components/AccessManagement/AccountTable.tsx:38]: Let me know if you still want to change it or keep it as it is.

Comment (katsukixyz) [reply web/components/AccessManagement/AccountTable.tsx:38]: I think we'll refactor this page so you can leave it like that - thanks!

**Review by katsukixyz — APPROVED** [review]
